DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs

```diff
@@ -20,39 +21,155 @@ public RegistrationDataService(IDbConnection connection)
             this.connection = connection;
         }
 
-        public string RegisterDelegate(DelegateRegistrationModel delegateRegistrationModel)
+        public string RegisterNewUserAndDelegateAccount(DelegateRegistrationModel delegateRegistrationModel)
         {
-            var values = new
+            // TODO HEEDLS-886: this method previously returned error codes as well as candidate numbers.
+            // any code that calls it and handled those errors on the basis of the codes needs to be updated
+            var userValues = new
+                {
+                    delegateRegistrationModel.FirstName,
+                    delegateRegistrationModel.LastName,
+                    delegateRegistrationModel.PrimaryEmail,
+                    delegateRegistrationModel.JobGroup,
+                    delegateRegistrationModel.Active,
+                    PasswordHash = "temp",
+                    ProfessionalRegistrationNumber = (string?)null,
+                };
+
+            var userId = connection.QuerySingle<int>(
+                @"INSERT INTO Users
+                    (
+                        PrimaryEmail,
+                        PasswordHash,
+                        FirstName,
+                        LastName,
+                        JobGroupID,
+                        ProfessionalRegistrationNumber,
+                        Active
+                    )
+                    OUTPUT Inserted.ID
+                    VALUES
+                    (
+                        @primaryEmail,
+                        @passwordHash,
+                        @firstName,
+                        @lastName,
+                        @jobGroup,
+                        @professionalRegistrationNumber,
+                        @active
+                    )",
+                    userValues
+                );
+
+            var initials = delegateRegistrationModel.FirstName.Substring(1) +
+                           delegateRegistrationModel.LastName.Substring(1);
+
+            string candidateNumber;
+            // this SQL is reproduced mostly verbatim from the uspSaveNewCandidate_V10 procedure in the legacy codebase.
+            connection.Open();

```

Comment:
It's necessary for the connection.BeginTransaction() invocation, but since I'm rebuilding it I'll see if I can get it to work using our more usual TransactionScope approach.

Reply:
TransactionScopes aren't substitutable with dapper Transactions, so I think we're stuck with connection.Open() for now.